### DISCUSSION TITLE
Update all but exclude teams and office

### DISCUSSION THREAD

**Discussion by drummbelbummel** [microsoft/winget-pkgs#124620] [Ideas]
I really love winget as it is making it very easy to keep non microsoft apps "up to date".
But every time i am struggling with teams and office.
It looks like winget is not able to update these applications in an environment with intune mdm.
So i wanted to use a script to exclude those two - just exporting "winget list --upgrade-available" and than looping through the result and skipping the id's
But - the comments in the output are not helpfull for what i want to do.
Could there be a parameter like "display only" with the option to get only the id or name of the package?
So i could loop through each app and decide what i want to happen.

**Comment by mdanish-kh** [comment]
Have you tried using the WinGet powershell module? https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/

There isn't an option in the PowerShell module [yet](https://github.com/microsoft/winget-cli/issues/3234) to exclude a package from update (what the [`winget pin`](https://learn.microsoft.com/en-us/windows/package-manager/winget/pinning), command available in the client offers), but you can get the desired result through a command like:

```powershell

Get-WinGetPackage | Where-Object {($_.IsUpdateAvailable -eq $true) -and ($_.Id -ne 'Microsoft.Teams') -and ($_.Id -ne 'Microsoft.Office')} | ForEach-Object { Update-WinGetPackage -Id $_.ID }
```

This will get all the packages available for update, filter out Microsoft.Teams and Microsoft.Office, and then pipe that to Update-WinGetPackage to upgrade all other packages

**Reply by Trenly** [reply]
Putting the "native" cli implementation here for reference;

Run once:
```
winget pin add --id Microsoft.Teams
winget pin add --id Microsoft.Office
```
After running the above, then `winget upgrade --all` will upgrade everything except Teams and Office

**Reply by Adi-Shuji** [reply]
Just sharing a comment I saw on how to unpin apps, in case you want to update Teams or Office using the command in the future from https://learn.microsoft.com/en-us/answers/questions/1203328/winget-exe-upgrade-all-exclude

Run:
```
winget pin remove --id Microsoft.Teams
winget pin remove --id Microsoft.Office
``` 

https://learn.microsoft.com/en-us/windows/package-manager/winget/pinning
